base/bucket.go

```diff
@@ -561,3 +595,35 @@ func IsCasMismatch(bucket Bucket, err error) bool {
 
 	return false
 }
+
+func GetMaxVbnoForBucket(bucket Bucket) uint16 {
+
+	// TODO: Move to sg-bucket interface (https://github.com/couchbase/sync_gateway/issues/2418)
+
+	var maxVbNo uint16 = 1024
+	var err error
+
+	switch bucket := bucket.(type) {
+	case CouchbaseBucketGoCB:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case *CouchbaseBucketGoCB:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case CouchbaseBucketGoCBGoCouchbaseHybrid:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case *CouchbaseBucketGoCBGoCouchbaseHybrid:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case CouchbaseBucket:
+		maxVbNo, err = bucket.GetMaxVbno()
+	case *CouchbaseBucket:
+		maxVbNo, err = bucket.GetMaxVbno()
+	default:
+		Warn("Not a couchbase bucket - assuming walrus, setting maxVbNo=1024")
+	}
```

Comment:
> Does having multiple options in a single case work, or does that break the usage of bucket inside the case?

I just tried it:

* [Multiple options in single case gives an error](https://play.golang.org/p/mJtnjwPm4T): `typedThing.Hello undefined (type interface {} is interface with no methods)`
* [Standalone case options works](https://play.golang.org/p/CRwHBBNTQ-)